### PR TITLE
Fixes Gyrotrons

### PR DESCRIPTION
**Pull request by PaulRitter**
Fixes gyrotrons and probably a bunch of other machines.

apparently alot of code calls dumbfire without a dir argument. huh

[bugfix]
grafik

### CODE REVIEW

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -124,6 +124,9 @@ var/list/beam_master = list()
 	shot_from = A //temporary
 
 /obj/item/projectile/beam/dumbfire(var/dir)
+	dir = dir ? dir : src.dir
```
Comment (Exxion) [inline code/modules/projectiles/projectile/beams.dm]:
```suggestion
	dir = dir || src.dir
```
It's a shortcut for the same thing, but marginally faster

File: code/modules/projectiles/projectile/beams.dm [hunk 2]
```diff
@@ -124,6 +124,9 @@ var/list/beam_master = list()
...
+	starting = starting ? starting : src.loc
```
Comment (Exxion) [inline code/modules/projectiles/projectile/beams.dm]:
```suggestion
	starting = starting || loc
```

**Comment by PaulRitter** [comment]
~~wait dont merge~~
good to go

**Comment by kernikov** [comment]
BASED.

**Comment by Exxion** [comment]
All those `src.` you added shouldn't be necessary except *maybe* the one on `src.dir` but I don't care enough to ask you to remove them

**Comment by PaulRitter** [comment]
im aware that only the src.dir matters, but i thought why not and just went all out

**Pull request merged**